govcd/openapi.go

```diff
@@ -689,6 +689,33 @@ func (client *Client) newOpenApiRequest(apiVersion string, params url.Values, me
 	return req
 }
 
+// OpenApiTestConnection Tests a connection, including SSL handshake and hostname verification.
+func (client *Client) OpenApiTestConnection(testConnection types.TestConnection) (*types.TestConnectionResult, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointTestConnection
+
+	minimumApiVersion, err := client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	returnTestConnectionResult := &types.TestConnectionResult{
+		TargetProbe: &types.ProbeResult{},
+		ProxyProbe:  &types.ProbeResult{},
+	}
+
+	err = client.OpenApiPostItem(minimumApiVersion, urlRef, nil, testConnection, returnTestConnectionResult, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating Edge Gateway: %s", err)
```

Comment:
It is not an edge gateway, is it?

Reply:
Copy/Paste issue. Solved and thanks for spotting that!